native/src/main/java/io/ballerina/stdlib/websub/NativeHttpToWebsubAdaptor.java

```diff
@@ -103,8 +104,16 @@ private static Object invokeRemoteFunction(Environment env, BObject bSubscriberS
         StrandMetadata metadata = new StrandMetadata(module.getOrg(), module.getName(), module.getVersion(),
                 parentFunctionName);
         Object[] args = new Object[]{message, true};
-        env.getRuntime().invokeMethodAsync(bSubscriberService, remoteFunctionName, null, metadata,
-                new SubscriberCallback(balFuture, module), args);
+        if (bSubscriberService.getType().isIsolated()
```

Comment:
Shouldn't we check only the latter condition? Because, bSubscriberService isolated means the remote functions of the particular service is implicitly isolated right. 
@MaryamZi

Reply:
An isolated (service) object can have non-isolated methods. So we need to check both.

An isolated object only guarantees safe access of the object's fields.

Reply:
So that means, we have to do the AND operation rather than the OR right

Reply:
This is fixed with [d3066c3b1a486db7364fa816dbe33aa2b5de030d](https://github.com/ballerina-platform/module-ballerina-websub/pull/421/commits/d3066c3b1a486db7364fa816dbe33aa2b5de030d)